### PR TITLE
[W4.4b][W11-A1] Sashankh Chengavalli Kumar

### PR DESCRIPTION
**Pull request by ChengSashankh**
Implemented classes 4.4b

### CODE REVIEW

File: src/seedu/addressbook/data/person/Address.java [hunk 1]
```diff
@@ -8,12 +8,21 @@
...
     public static final String ADDRESS_VALIDATION_REGEX = ".+";
+    public static final String ADDRESS_SPLIT_REGEX = ",";
+    public static final int BLOCK_ADDRESS_INDEX = 0;
+    public static final int STREET_ADDRESS_INDEX = 1;
+    public static final int UNIT_ADDRESS_INDEX = 2;
+    public static final int POSTAL_CODE_ADDRESS_INDEX = 3;
 
     public final String value;
     private boolean isPrivate;
+    Block block;
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java:22]: While this is alright, try to make it private. 
In general, start off with private members, then use protected or public only if absolutely necessary.

File: src/seedu/addressbook/data/person/Address.java [hunk 2]
```diff
@@ -8,12 +8,21 @@
  */
 public class Address {
 
-    public static final String EXAMPLE = "123, some street";
+    public static final String EXAMPLE = "123, some street, some unit, some postal code";
     public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
     public static final String ADDRESS_VALIDATION_REGEX = ".+";
+    public static final String ADDRESS_SPLIT_REGEX = ",";
+    public static final int BLOCK_ADDRESS_INDEX = 0;
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java:15]: 2 things 
- Good to see you have eliminated magic numbers in the code
- However, please take a look at the coding standards and find out what is the correct convention for naming constants

File: src/seedu/addressbook/data/person/Address.java [hunk 3]
```diff
@@ -22,11 +31,77 @@
...
+        unit = new Unit(getUnitFromAddress(addressInParts, numberOfParts));
+        postalCode = new PostalCode(getPostalCodeFromAddress(addressInParts, numberOfParts));
+    }
+
+    /**
+     * Returns the postal code from address
+     *
+     * @param addressInParts
+     */
+    private String getPostalCodeFromAddress(String[] addressInParts, int numberOfParts){
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java:53]: These methods are well thought of!

File: src/seedu/addressbook/data/person/Block.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package seedu.addressbook.data.person;
+
+public class Block {
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Block.java:3]: This class and its non-trivial methods should have header comments. 
If not, developers who write client code for this class will not know how to use it.

File: src/seedu/addressbook/data/person/Address.java [hunk 5]
```diff
@@ -22,11 +31,77 @@
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
         String trimmedAddress = address.trim();
+        int numberOfParts;
+        this.value = new String(trimmedAddress);
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java:35]: Why is the `value` still being retained? Consider all the mishaps that can happen due to data duplication!

Also, consider moving the validation into the new classes you created.

**Review by okkhoy — COMMENTED** [review]
Please note that you need to update the documentation (user guide and in program help) and the tests in addition to making the code change.

**Comment by okkhoy** [comment]
@ChengSashankh 
Some comments added. Please ack & close the PR after reading comments.

**Comment by ChengSashankh** [comment]
Acknowledged. Thank you for the comments.

**Pull request closed without merging**